### PR TITLE
update create detector page and detector detail page to add custom result index lifecycle management settings

### PR DESCRIPTION
**Pull request by jackiehanyang**
### Description

update create detector page and detector detail page to add custom result index lifecycle management settings

https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/assets/41348518/eae9a505-f272-4608-8c80-271b9292f984

### Issues Resolved

https://github.com/opensearch-project/anomaly-detection/issues/1213

### Check List

- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/DefineDetector/components/CustomResultIndex/CustomResultIndex.tsx [hunk 1]
```diff
@@ -115,6 +118,121 @@ function CustomResultIndex(props: CustomResultIndexProps) {
...
+                error={getError(field.name, form)}
+              >
+                <EuiFlexGroup gutterSize="s" alignItems="center">
+                  <EuiFlexItem grow={false}>
+                    <EuiFieldNumber
+                      name="resultIndexMinSize"
+                      id="resultIndexMinSize"
+                      placeholder="Max index size"
+                      data-test-subj="resultIndexMinSize"
+                      min={1}
```
Comment (kaituo) [inline public/pages/DefineDetector/components/CustomResultIndex/CustomResultIndex.tsx:183]: Can we change the min to 1000 or change the unit to GB? 1MB is too small.

File: public/pages/ReviewAndCreate/components/DetectorDefinitionFields/DetectorDefinitionFields.tsx [hunk 2]
```diff
@@ -217,6 +217,24 @@ export const DetectorDefinitionFields = (
             description={get(props, 'detector.resultIndex', '-')}
           />
         </EuiFlexItem>
+        <EuiFlexItem>
+          <ConfigCell
+            title="Custom result index min age"
+            description={get(props, 'detector.resultIndexMinAge', '-')}
```
Comment (kaituo) [inline public/pages/ReviewAndCreate/components/DetectorDefinitionFields/DetectorDefinitionFields.tsx:223]: Can you add unit in the following 3 items?

File: public/pages/DefineDetector/components/CustomResultIndex/CustomResultIndex.tsx (hunk 1 above)
Comment (jackiehanyang) [reply public/pages/DefineDetector/components/CustomResultIndex/CustomResultIndex.tsx]: updated

File: public/pages/ReviewAndCreate/components/DetectorDefinitionFields/DetectorDefinitionFields.tsx (hunk 2 above)
Comment (jackiehanyang) [reply public/pages/ReviewAndCreate/components/DetectorDefinitionFields/DetectorDefinitionFields.tsx]: updated

**Review by kaituo — APPROVED** [review]

**Pull request merged**